lib/commands/location.js

```diff
@@ -31,10 +31,13 @@ export default {
     // '3' is 'Always' in the privacy
     // https://developer.apple.com/documentation/corelocation/clauthorizationstatus
     if (authorizationStatus !== AuthorizationStatus.authorizedAlways) {
-      this.log.errorAndThrow(
+      throw this.log.errorWithException(
         `Location service must be set to 'Always' in order to ` +
-          `retrive the current geolocation data. Please set it up manually via ` +
-          `'Settings > Privacy > Location Services -> WebDriverAgentRunner-Runner'`,
+          `retrieve the current geolocation data. Please set it up manually via ` +
+          `'Settings > Privacy > Location Services -> WebDriverAgentRunner-Runner'. ` +
+          `Newer simulator environment may not have the settings item. Please use ` +
```

Comment:
As far as I can understand this is an Apple bug, which could be fixed later. Shall we explicitly mention that?

Reply:
good point. hm, maybe here can drop _Newer simulator environment may not have the settings item._.